[stdlib] Documentation revisions

natecook1000 opened the pull request:
- Update NSRange -> Range guidance
- Fix example in Optional
- Improve RangeExpression docs
- Fix issue in UnsafeRawBufferPointer.initializeMemory
- Code point -> scalar value most places
- Reposition the dot above the scripty `i'
- Fix ExpressibleByArrayLiteral code sample

natecook1000 committed:
[stdlib] Documentation revisions

- Update NSRange -> Range guidance
- Fix example in Optional
- Improve RangeExpression docs
- Fix issue in UnsafeRawBufferPointer.initializeMemory
- Code point -> scalar value most places
- Reposition the dot above the scripty `i'
- Fix ExpressibleByArrayLiteral code sample

natecook1000 commented:
@swift-ci Please smoke test

xwu commented on stdlib/public/core/Range.swift.gyb:
```diff
@@ -10,11 +10,14 @@
+/// A type that conforms to `RangeExpression` can convert itself to a
+/// `Range<Bound>` of indices within a given collection.
 public protocol RangeExpression {
+  /// The type that an expression describes a range of.
```
Suggestion: how about the following?

> The type of elements for which the expression describes a range.

xwu commented on stdlib/public/core/UnicodeScalar.swift:
```diff
@@ -109,39 +109,39 @@ extension Unicode.Scalar :
   ///     }
+  ///
+  /// - Parameter v: The Unicode code point to use for the scalar. The
+  ///   initializer succeeds if `v` is a valid Unicode scalar value, in the
```
Nit: no comma.

natecook1000 committed:
[stdlib] Additional revisions

- Incorporate feedback from @xwu
- Add parameters for `dump`
- Duplicate some missing integer comments

natecook1000 commented:
@swift-ci Please smoke test and merge
